tests/unit/Promise.ts

```diff
@@ -76,19 +70,33 @@ export function addPromiseTests(suite: any, Promise: PromiseType) {
 			}));
 		},
 
-		'sparse array': function () {
-			let dfd = this.async();
-			let iterable: any[] = [];
+		'sparse array': {
+			all() {
+				let dfd = this.async();
+				let iterable: any[] = [];
 
-			iterable[0] = Promise.resolve(0);
-			iterable[3] = Promise.resolve(3);
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
 
-			Promise.all(iterable).then(dfd.callback(function (value: number[]) {
-				assert.strictEqual(value[0], 0);
-				assert.isUndefined(value[1]);
-				assert.isUndefined(value[2]);
-				assert.strictEqual(value[3], 3);
-			}));
+				Promise.all(iterable).then(dfd.callback(function (value: number[]) {
+					assert.strictEqual(value[0], 0);
+					assert.isUndefined(value[1]);
+					assert.isUndefined(value[2]);
+					assert.strictEqual(value[3], 3);
+				}));
+			},
+
+			race() {
+				let dfd = this.async();
+				let iterable: any[] = [];
+
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
+
+				Promise.race(iterable).then(dfd.callback(function (value: number) {
+					assert.strictEqual(value, 0);
```

Comment:
I may just be uninformed and paranoid here, but is this a sufficient test?

* The array contains two already-resolved promises, separated by two unpopulated indices
* The first item (before any undefined indices) is the answer anyway

I suppose `race` is still going to iterate over the entire array.  What would be the behavior of this if we didn't handle sparse arrays though?  Would this assertion fail or would the entire operation throw or hit the errback (which we're not handling in this test)?